kythe/rust/fuchsia_extractor/README.md

```diff
@@ -1,9 +1,173 @@
-# Fuchsia Rust compilation extractor
+# A compilation extractor for Rust, for Fuchsia OS
 
-This directory contains a Rust compilation extractor for the [Fuchsia OS][fx]
+This directory contains a Rust compilation extractor made specifically for the
+[Fuchsia OS][fx].  It can not be applied in general to a different GN build,
+since it requries Fuchsia-specific compiler instrumentation.
+
+At the moment, the split between the build-system specific work and the
+language-specific work is not as clean as it could be.  This is for two main
+reasons.  First and foremost, the Rust compiler toolchain is not yet factored
+to allow for easy reuse of the components that an indexer would need. Second,
+the build system at the moment does not produce a compilation extraction that
+an ideal Rust extractor can use.  As work is underway to improve both of these
+two issues, we decided to go with a pragmatic approach first, and once the
+libraries are available, we can update the indexer and the extractor to match.
 
 [fx]: https://www.fuchsia.dev
+[gn]: https://gn.googlesource.com/gn/
 
-Fuchsia build system is a bit specific, and is custom built on top of [GN][gn].
+Technically, this extractor produces `kzip` files as any other extractor does,
+and the `kzip` file specification is well known.  However, in practice, the
+kzip specification is open for interpretation (for example, what if you need
+to inject a specific extra file to a compilation unit?).  That in turn can
```

Comment:
By putting it in the compilation unit's `required_input` list.

Reply:
Ack.